saito-analytics/src/runner.rs

```diff
@@ -0,0 +1,236 @@
+        let keys = generate_keys();
+        let wallet = Wallet::new(keys.1, keys.0);
+        let _public_key = wallet.public_key.clone();
+        let _private_key = wallet.private_key.clone();
+        let peers = Arc::new(RwLock::new(PeerCollection::new()));
+        let wallet_lock = Arc::new(RwLock::new(wallet));
+        let blockchain = Arc::new(RwLock::new(Blockchain::new(wallet_lock.clone())));
+        let mempool = Arc::new(RwLock::new(Mempool::new(wallet_lock.clone())));
+
+        let (sender_to_miner, receiver_in_miner) = tokio::sync::mpsc::channel(10);
+        let configs = Arc::new(RwLock::new(TestConfiguration {}));
+
+        Self {
+            wallet_lock: wallet_lock.clone(),
+            blockchain,
+            mempool,
+            latest_block_hash: [0; 32],
+            network: Network::new(
+                Box::new(TestIOHandler::new()),
+                peers.clone(),
+                wallet_lock.clone(),
+                configs.clone(),
+            ),
+            peers: peers.clone(),
+            storage: Storage::new(Box::new(TestIOHandler::new())),
+            sender_to_miner: sender_to_miner.clone(),
+            receiver_in_miner,
+            configs,
+        }
+    }
+
+    //load blocks via id
+    //this is just the vector of blocks
+    pub async fn get_blocks_vec(&self) -> Vec<Block> {
+        let mut blocks = Vec::new();
+
+        let (blockchain, _blockchain_) = lock_for_read!(self.blockchain, LOCK_ORDER_BLOCKCHAIN);
+
+        let latest_id = blockchain.get_latest_block_id();
+        for i in 1..=latest_id {
+            let block_hash = blockchain
+                .blockring
+                .get_longest_chain_block_hash_by_block_id(i as u64);
+            //println!("WINDING ID HASH - {} {:?}", i, block_hash);
+            let block = blockchain.get_block(&block_hash).unwrap().clone();
+            blocks.push(block);
+        }
+
+        blocks
+    }
+
+    pub async fn load_blocks_from_path(&mut self, directory_path: &str) {
+        //TODO put util in core/storage or use existing one here
+        let blocks_result = load_blocks_disk(&directory_path);
+
+        blocks_result.as_ref().unwrap_or_else(|e| {
+            eprintln!("Error reading blocks: {}", e);
+            std::process::exit(1);
+        });
+        {
+            let blocks = blocks_result.unwrap();
+            let (mut mempool, _mempool_) = lock_for_write!(self.mempool, LOCK_ORDER_MEMPOOL);
+
+            debug!("got blocks {}", blocks.len());
+            for mut block in blocks {
+                block.force_loaded = true;
+                block.generate();
+                debug!("block : {:?} loaded from disk", hex::encode(block.hash));
+                mempool.add_block(block);
+            }
+        }
+
+        let (mut blockchain, _blockchain_) =
+            lock_for_write!(self.blockchain, LOCK_ORDER_BLOCKCHAIN);
+
+        let (configs, _configs_) = lock_for_read!(self.configs, LOCK_ORDER_CONFIGS);
```

Comment:
CONFIGs lock should be locked before BLOCKCHAIN lock. otherwise might cause deadlocks. (not for this specific code since it's for a tool. but better to think about those also)